source/s2n/s2n_tls_channel_handler.c

```diff
@@ -717,6 +719,38 @@ static int s_parse_protocol_preferences(
     return AWS_OP_SUCCESS;
 }
 
+static size_t s_tl_cleanup_key = 0; /* Address of variable serves as key in hash table */
+static struct aws_event_loop_local_object s_tl_cleanup_object;
+
+static void s_aws_cleanup_s2n_thread_local_state(void *user_data) {
+    (void)user_data;
+
+    aws_tls_clean_up_thread_local_state();
+}
+
+static int s_s2n_tls_channel_handler_schedule_tl_cleanup(struct aws_channel_slot *slot) {
```

Comment:
Add comment explaining what this is all about. The "tl" acronym isn't super obvious in a "tls" file.
/* s2n allocates many thread-local datastructures. We need to clean these up when the even-loop's thread exits. */